modules/core/src/com/haulmont/cuba/security/app/EntityAttributeChanges.java

```diff
@@ -14,38 +14,71 @@
  * limitations under the License.
  */
 
-package com.haulmont.cuba.core.sys.persistence;
+package com.haulmont.cuba.security.app;
 
+import com.haulmont.cuba.core.entity.Entity;
+import org.eclipse.persistence.descriptors.changetracking.ChangeTracker;
+import org.eclipse.persistence.internal.descriptors.changetracking.AttributeChangeListener;
 import org.eclipse.persistence.sessions.changesets.AggregateChangeRecord;
 import org.eclipse.persistence.sessions.changesets.ChangeRecord;
 import org.eclipse.persistence.sessions.changesets.ObjectChangeSet;
 
 import javax.annotation.Nullable;
-import java.util.*;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 /**
- * INTERNAL. Accumulates changes in entity attributes.
+ * Accumulates changes in entity attributes.
```

Comment:
Accumulates changes in entity attributes.
The {@code #addChanges(Entity entity)} method is used by EntityLog when registering changes for the whole entity automatically. 
You can add changes for some attributes by calling {@link EntityAttributeChanges#addChange(String, Object)} and calling EntityLog programmatically.